src/create-http-client.ts

```diff
@@ -20,136 +21,140 @@ const HOST_REGEX = /^(?!\w+:\/\/)([^\s:]+\.?[^\s:]+)(?::(\d+))?(?!:)$/
+    if (config.host && HOST_REGEX.test(config.host)) {
+        const parsed = config.host.split(':')
+        if (parsed.length === 2) {
+            ;[hostname, port] = parsed
+        } else {
+            hostname = parsed[0]
+        }
+    }
+
+    // Ensure that basePath does start but not end with a slash
+    if (config.basePath) {
+        config.basePath = `/${config.basePath.split('/').filter(Boolean).join('/')}`
+    }
+
+    const baseURL =
+        options.baseURL || `${protocol}://${hostname}:${port}${config.basePath}/spaces/${space}`
+
+    if (!config.headers.Authorization && typeof config.accessToken !== 'function') {
+        config.headers.Authorization = 'Bearer ' + config.accessToken
+    }
+
+    // Set these headers only for node because browsers don't like it when you
+    // override user-agent or accept-encoding.
+    // The SDKs should set their own X-Contentful-User-Agent.
+    if (isNode()) {
+        config.headers['user-agent'] = 'node.js/' + getNodeVersion()
+        config.headers['Accept-Encoding'] = 'gzip'
+    }
+
+    const axiosOptions = {
+        // Axios
+        baseURL,
+        headers: config.headers,
+        httpAgent: config.httpAgent,
+        httpsAgent: config.httpsAgent,
+        paramsSerializer: qs.stringify,
+        proxy: config.proxy,
+        timeout: config.timeout,
+        adapter: config.adapter,
+        maxContentLength: config.maxContentLength,
+        maxBodyLength: config.maxBodyLength,
+        // Contentful
+        logHandler: config.logHandler,
+        responseLogger: config.responseLogger,
+        requestLogger: config.requestLogger,
+        retryOnError: config.retryOnError,
+    }
+    const instance = axios.create(axiosOptions) as AxiosInstance
+    instance.httpClientParams = options
+
+    /**
+     * Creates a new axios instance with the same default base parameters as the
+     * current one, and with any overrides passed to the newParams object
+     * This is useful as the SDKs use dependency injection to get the axios library
+     * and the version of the library comes from different places depending
+     * on whether it's a browser build or a node.js build.
+     * @private
+     * @param {CreateHttpClientParams} httpClientParams - Initialization parameters for the HTTP client
+     * @return {ContentfulAxiosInstance} Initialized axios instance
+     */
+    instance.cloneWithNewParams = function (
+        newParams: Partial<CreateHttpClientParams>
+    ): AxiosInstance {
+        return createHttpClient(axios, {
+            ...copy(options),
+            ...newParams,
+        })
+    }
+
+    /**
+     * Apply interceptors.
+     * Please note that the order of interceptors is important
+     */
+
+    if (config.onBeforeRequest) {
+        instance.interceptors.request.use(config.onBeforeRequest)
+    }
+
+    if (typeof config.accessToken === 'function') {
+        asyncToken(instance, config.accessToken)
+    }
+
+    if (config.throttle !== 0) {
```

Comment:
You probably want to check with `if (config.throttle)` here. In case the user defines `options = { throttle: undefined }` you currently would still include the throttle interceptor.